src/compiler/moduleSpecifiers.ts

```diff
@@ -11,6 +11,8 @@ namespace ts.moduleSpecifiers {
         readonly ending: Ending;
     }
 
+    let discoverProbableSymlinksCached: (() => ReadonlyMap<string>) | undefined;

```

Comment:
I don't like that we are caching here.. It should be on host somehow and not in global ts scope. We need to pass around information wherever it needs to instead.

Reply:
This is a minor implementation detail that I’m not sure the host needs to know about. I also don’t love that it’s stored on the global scope, but if you look at `withCachedSymlinks`, it only gets set for the duration of a callback. I can try to move the cache inside `withCachedSymlinks` and pass them around some of the other worker functions.

Reply:
There's probably a good argument to be made for wrapping the functionality the `moduleSpecifiers` namespace provides and passing it into the checker on a host object via `Program`, if only so the `Program` (or whoever is hosting the program) can manage the caches.

Reply:
I added an optional member `getProbableSymlinks` to `ModuleSpecifierResolutionHost` and made `LanguageServiceHost` extend that (it was already assignable to it) in https://github.com/microsoft/TypeScript/pull/32517/commits/a9b63697997d64777674f4a42ecc4063b7f5a8ac. Let me know what you think.